web-admin/src/features/projects/status/ProjectResources.svelte

```diff
@@ -2,73 +2,166 @@
+    type V1Resource,
+    type V1ListResourcesResponse,
+    createRuntimeServiceListResources,
   } from "@rilldata/web-common/runtime-client";
   import { runtime } from "@rilldata/web-common/runtime-client/runtime-store";
   import { useQueryClient } from "@tanstack/svelte-query";
   import Button from "web-common/src/components/button/Button.svelte";
   import ProjectResourcesTable from "./ProjectResourcesTable.svelte";
-  import RefreshConfirmDialog from "./RefreshConfirmDialog.svelte";
+  import RefreshAllSourcesAndModelsConfirmDialog from "./RefreshAllSourcesAndModelsConfirmDialog.svelte";
+  import { ResourceKind } from "@rilldata/web-common/features/entity-management/resource-selectors";
+  import { eventBus } from "@rilldata/web-common/lib/event-bus/event-bus";
 
   const queryClient = useQueryClient();
   const createTrigger = createRuntimeServiceCreateTrigger();
 
+  let isConfirmDialogOpen = false;
+  let currentResourceName: string | undefined;
   let isReconciling = false;
-  let isRefreshConfirmDialogOpen = false;
+
+  const INITIAL_REFETCH_INTERVAL = 500; // Start at 500ms
+  const MAX_REFETCH_INTERVAL = 10_000; // Cap at 10s
+  const BACKOFF_FACTOR = 2; // Double each time
+  let currentRefetchInterval = INITIAL_REFETCH_INTERVAL;
 
   $: ({ instanceId } = $runtime);
 
-  $: resources = createRuntimeServiceListResources(
-    instanceId,
-    // All resource "kinds"
-    undefined,
-    {
-      query: {
-        select: (data) => {
-          // Filter out the "ProjectParser" resource
-          return data.resources.filter(
-            (resource) =>
-              resource.meta.name.kind !== "rill.runtime.v1.ProjectParser",
-          );
-        },
-        refetchOnMount: true,
-        refetchOnWindowFocus: true,
-        refetchInterval: isReconciling ? 500 : false,
-      },
-    },
-  );
+  function isResourceErrored(resource: V1Resource) {
+    return !!resource.meta.reconcileError;
+  }
 
-  $: isAnySourceOrModelReconciling = $resources?.data?.some(
-    (resource) =>
+  function isResourceReconciling(resource: V1Resource) {
+    return (
       resource.meta.reconcileStatus ===
         V1ReconcileStatus.RECONCILE_STATUS_PENDING ||
       resource.meta.reconcileStatus ===
-        V1ReconcileStatus.RECONCILE_STATUS_RUNNING,
-  );
+        V1ReconcileStatus.RECONCILE_STATUS_RUNNING
+    );
+  }
 
-  function refreshAllSourcesAndModels() {
-    isReconciling = true;
+  $: resources = createRuntimeServiceListResources(instanceId, undefined, {
+    query: {
+      select: (data: V1ListResourcesResponse) => ({
+        ...data,
+        // Filter out project parser and refresh triggers
+        resources: data?.resources?.filter(
+          (resource: V1Resource) =>
+            resource.meta.name.kind !== ResourceKind.ProjectParser &&
+            resource.meta.name.kind !== ResourceKind.RefreshTrigger,
+        ),
+      }),
+      refetchInterval: (data) => {
+        if (
+          $resources?.isError ||
```

Comment:
This still isn't a pure function, given `$resources.isError` is still here. I'm worried about a circular reference.